### ISSUE TITLE
Make flatpak compatible with scientific packages thar run in parallel

### ISSUE THREAD

**Issue by HenriqueCSJ** [flatpak/flatpak#3162]
Flatpak is a giant step in the right direction and the possibility of a secure, distro agnostic package is something that can potentially make the life of the scientific community much easier. It is nice to know that with great distros like Fedora gradually moving to flatpak, soon or later my favorite scientific desktop software is going to be ported. But, at this point and as far as I know, the CLI software (like the quantum chemistry packages) cant be packaged as flatpack because they rely on libraries dependant on openmpi/mpich and execution in parallel.

So, my request is: make flatpaks possible for software that runs in parallel too.

**Comment by alexlarsson** [comment]
What is special about openmpi/mpich that needs special runtime support? Its just some software, no?

**Comment by HenriqueCSJ** [comment]
No. It does not look that simple: https://src.fedoraproject.org/rpms/boost/pull-request/1

**Comment by alexlarsson** [comment]
I don't understand, there is nothing in those references but packaging issues.

**Comment by HenriqueCSJ** [comment]
Hi, Alex, I'm not able to comment in detail because I'm not a developer, so mine is a layman point of view on the issue.
I believe that because Message Passing Interface (MPI) packages require root privileges to be installed (since they enable parallel computing) they can't be satisfied as a containerized dependency.
Does that make any sense? And, if it does make sense, is this one of those cases where flatpak is not designed for?

**Comment by mwleeds** [comment]
Looks like there's more specific information here: https://src.fedoraproject.org/rpms/boost/pull-request/1#comment-0
